Classes/Command/GlossaryCommandController.php

```diff
@@ -0,0 +1,166 @@
+        $languagesWithUpdates = $this->glossaryEntryRepository->findLanguagesThatRequireSyncing($lastExecutedAt);
+        if (count($languagesWithUpdates) === 0) {
+            return;
+        }
+
+        // fetching the required language pairs can extend the languages we have to sync
+        [$languagePairs, $languagesToSync] = $this->deepLApi->getLanguagePairs($languagesWithUpdates);
+
+        // aggregate entries
+        $aggregates = [];
+        $entries = $this->glossaryEntryRepository->findByLanguages($languagesToSync);
+        foreach ($entries as $entry) {
+            $identifier = $entry->aggregateIdentifier;
+            if (!array_key_exists($entry->aggregateIdentifier, $aggregates)) {
+                $aggregates[$identifier] = [];
+            }
+            $aggregates[$identifier][$entry->glossaryLanguage] = $entry->text;
+        }
+
+        // build glossary entries in DeepL format
+        $glossaries = [];
+        foreach ($aggregates as $aggregate) {
+            foreach ($languagePairs as $languagePair) {
+                $source = $languagePair['source'];
+                $target = $languagePair['target'];
+                if (array_key_exists($source, $aggregate) && array_key_exists($target, $aggregate)) {
+                    $sourceText = trim($aggregate[$source]);
+                    $targetText = trim($aggregate[$target]);
+                    if (!empty($sourceText) && !empty($targetText)) {
+                        $internalGlossaryKey = $this->deepLApi->getInternalGlossaryKey($source, $target);
+                        if (!array_key_exists($internalGlossaryKey, $glossaries)) {
+                            $glossaries[$internalGlossaryKey] = [];
+                        }
+                        $entry = $sourceText . self::GLOSSARY_ENTRY_SEPARATOR . $targetText;
+                        $glossaries[$internalGlossaryKey][] = $entry;
+                    }
+                }
+            }
+        }
+
+        $this->updateDeepLGlossaries($glossaries);
+
+        $this->storage->set('lastExecutionTimestamp', $currentTime);
+        $this->storage->set('forceCompleteSync', false);
+    }
+
+    /**
+     * DeepL glossaries are immutable,
+     * so we need to delete existing glossaries and create them afterwards.
+     * @throws ClientExceptionInterface
+     */
+    protected function updateDeepLGlossaries(array $newGlossaries): void
+    {
+        foreach ($newGlossaries as $internalGlossaryKey => $entries) {
+
+            if ($this->doesGlossaryExistInDeepL($internalGlossaryKey)) {
+                $this->deepLApi->deleteGlossary($this->glossaryKeyMapping[$internalGlossaryKey]);
+            }
+
+            [$sourceLangauge, $targetLangauge] = $this->deepLApi->getLanguagesFromInternalGlossaryKey($internalGlossaryKey);
+            $createData = [
+                'name' => "Solarwatt Website, source $sourceLangauge, target $targetLangauge",
```

Comment:
This seems to be uninteded from the original author ;-), needs to be replaced by the Site name.